pulp_docker/app/authorization.py

```diff
@@ -0,0 +1,172 @@
+            serialization.Encoding.DER,
+            serialization.PublicFormat.SubjectPublicKeyInfo
+        )
+        return key_in_der_format
+
+    def _split_into_encoded_groups(self, encoded_base32):
+        """Split encoded and truncated base32 into 12 groups separated by ':'."""
+        kid = encoded_base32[:4]
+        for index, char in enumerate(encoded_base32[4:], start=0):
+            if index % 4 == 0:
+                kid += ':' + char
+            else:
+                kid += char
+        return kid
+
+    def determine_access(self, user, scope):
+        """
+        Determine access permissions for a corresponding user.
+
+        This method determines whether the user has a valid access permission or not.
+        The determination is based on role based access control. For now, the access
+        is given out to anybody because the role based access control is not implemented
+        yet.
+
+        Args:
+            user (str): A name of the user who is trying to access a registry.
+            scope (str): A requested scope.
+
+        Returns:
+            list: An intersected set of the requested and the allowed access.
+
+        """
+        typ, name, actions = scope.split(':')
+        actions_list = actions.split(',')
+        permitted_actions = list(set(actions_list).intersection(['pull']))
+        return {'type': typ, 'name': name, 'actions': permitted_actions}
+
+    def _generate_claim_set(self, issuer, issued_at, subject, audience, access):
+        token_id = str(uuid.UUID(int=random.getrandbits(128), version=4))
+        expiration = issued_at + TOKEN_EXPIRATION_TIME
+        return {
+            'access': access,
+            'aud': audience,
+            'exp': expiration,
+            'iat': issued_at,
+            'iss': issuer,
+            'jti': token_id,
+            'nbf': issued_at,
+            'sub': subject
+        }
+
+
+class TokenRequestQueries:
+    """A data class that holds data retrieved from the request queries."""
+
+    def __init__(self, scope, service):
+        """
+        Store a scope and a service.
+
+        Args:
+            scope (str): A requested scope.
+            service (str): A service that request a Bearer token.
+
+        """
+        self.scope = scope
+        self.service = service
+
+    @classmethod
+    def init_from(cls, request):
+        """
+        Initialize the actual class with data retrieved from the request queries.
+
+        In this method, a validity and a presence of required queries (scope, service)
+        is checked as well.
+        """
+        try:
+            scope = request.query['scope']
+        except KeyError:
+            raise web.HTTPBadRequest(reason='A scope was not provided.')
+
+        try:
+            service = request.query['service']
+        except KeyError:
+            raise web.HTTPBadRequest(reason='A service name was not provided.')
+        if service not in KNOWN_SERVICES:
```

Comment:
why did we call service_name v2?

Reply:
Because the only known service is v2 https://github.com/pulp/pulp_docker/blob/59d93c8b30c071bdf68919524f916b83e47ee037/pulp_docker/app/content.py#L8. The service which can request a Bearer token

Reply:
do you know why in docker auth it is 'docker.registry.io'? we have same endpoints as docker registry

Reply:
Ah, snap! Yes, then I have to use something like 'localhost:24816' (or CONTENT_HOST constant itself).